src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,405 @@
+            case ("T"):
+                fw.write("T | " + taskStatus + " |" + task.description);
+                break;
+            case ("D"):
+                Deadline deadlineTask = (Deadline) task;
+                fw.write("D | " + taskStatus + " |" + task.description + "|" + deadlineTask.getBy());
+                break;
+            case ("E"):
+                Event eventTask = (Event) task;
+                fw.write("E | " + taskStatus + " |" + task.description + "|" + eventTask.getDuration());
+                break;
+            default:
+                System.out.println("Something went wrong!");
+                break;
+            }
+            fw.write(System.lineSeparator());
+        }
+        fw.close();
+    }
+    // Adapted from https://stackoverflow.com/questions/28947250/create-a-directory-if-it-does-not-exist-and-then-create-the-files-in-that-direct
+    public static void checkDirectoryExists(String filePath) {
+        File directory = new File(filePath);
+        if (!directory.exists()) {
+            directory.mkdir();
+        }
+    }
+
+    public static void saveTaskFile(String filePath) {
+        checkDirectoryExists(filePath);
+        try {
+            writeToFile(filePath);
+        } catch (IOException e) {
+            System.out.println("Something went wrong: " + e.getMessage());
+        }
+    }
+
+    // function populates empty task ArrayList given contents of the Task file
+    public static void fillArrayListFromFile(String input) {
+        String[] taskContent = input.split( "\\|");
+        String taskType = taskContent[0].trim();
+        String ifDone = taskContent[1].trim();
+        String taskName = taskContent[2];
+
+        switch (taskType) {
+        case ("T"):
+            tasks.add(new Todo(taskName));
+            break;
+        case ("D"):
+            String taskDeadline = taskContent[3];
+            tasks.add(new Deadline(taskName, taskDeadline));
+            break;
+        case ("E"):
+            String taskDuration = taskContent[3];
+            tasks.add(new Event(taskName, taskDuration));
+            break;
+        default:
+            System.out.println("Something went wrong!");
+            break;
+        }
+        if (ifDone.equals("1")) {
+            tasks.get(tasks.size() - 1).markAsDone();
+        }
+    }
+
+
+    public static void readTaskFile(String filePath) throws FileNotFoundException {
+        File taskFile = new File(filePath);
+        Scanner scan = new Scanner(taskFile);
+        while (scan.hasNext()) {
+            String taskToAdd = scan.nextLine();
+            fillArrayListFromFile(taskToAdd);
+        }
+    }
+
+    public static void loadTaskFile(String filePath) {
+        try {
+            readTaskFile(filePath);
+        } catch (FileNotFoundException e) {
+            System.out.println("File not found");
+        }
+    }
+
+    public static void main(String[] args) {
+        loadTaskFile("data/duke.txt");
```

Comment:
You could even have a huge try catch block here so that you could not need it in other places (making the code cleaner).